BR: Add a new doc about creating table in batch (#8659)
pingcap/docs-cn: br/br-batch-create-table.md

---
title: BR 数据恢复批量建表
summary: 了解 BR 数据恢复批量建表功能，在集群 restore 的情况下，BR 通过批量建表加速数据恢复。
---

# BR 批量建表 <span class="version-mark">从 v6.0 版本开始引入</span>

在 TiDB v6.0.0 之前，使用 BR 工具进行恢复任务时会先做建库建表操作，而后进行数据恢复。在建表过程中，BR 调用 TiDB 接口，类似于 SQL 的 create table。对于恢复中大量的建表， 每个 SQL Create Table 由 TiDB ddl owner 依次串行执行。这导致表比较多的场景中，建表时间过长。为了减少恢复任务建表时间过长，从 TiDB v6.0.0 起，BR 引入了恢复批量建表功能，此功能会默认开启。BR 可以通过该功能加速建表，从而减少数据恢复的时间。

## 使用场景

该功能默认打开，在表数量较多的时候，数据恢复效果较为明显。

## 使用方法

数据恢复批量建表功能默认打开，并默认设置 `--ddl-batch-size=128` 来恢复数据，无需额外配置。该参数在 restore 阶段并发的批量创建表。如需要关闭此功能，只需要设置 `--ddl-batch-size=0`， BR 会以老的方式来串行建表。

在 restore 数据中存在大量的表（如 > 5w），推荐增加并发参数 `--concurrency=1024` 与 `--ddl-batch-size=256` 一起使用，示例如下：

{{< copyable "shell-regular" >}}

```shell
bin/br restore table --db batchmark --table order_line -s local:///br_data/ --pd 172.16.5.198:2379 --log-file restore-concurrency.log --concurrency 1024 --ddl-batch-size=256
```

> * 此参数在数据恢复时，可以加速建表速度，测试显示6万张表恢复建表时间仅需5分钟

Reviewer: ```suggestion
* 此参数在数据恢复时，可以加速建表速度，测试显示6万张表恢复建表时间仅需5分钟 *
```

Reviewer: 是要加粗吗

## 使用限制

数据恢复批量建表是默认以一次128张表为批，并发批量建表的方案。且它需要 TiDB 和 BR 都大于 v6.0.0 版本。如果 TiDB 或者 BR 同时或者任意一方版本低于 v6.0.0 版本，BR 数据恢复会回退到老的建表方式，依次串行建表。

该功能的已知问题及其解决方案如下：

- BR 恢复存档时报错 `entry too large, the max entry size is 6291456, the size of data is 7690800`。

>     - 解决方法：可以尝试降低并发大小，如设置 `--ddl-batch-size=128` 或者更小的值。当你已经设置了 `--ddl-batch-size` 参数值后，在使用 BR 恢复数据时，BR 会先调用内部接口 `BatchCreateTableWithInfo`，然后使用内部 DDL job `ActionCreateTables`，最后根据已设置的 `--ddl-batch-size` 的值来在 TiDB 建表。当 TiDB 执行这个 DDL job 时，TiDB 会把 job 队列写到 TiKV 上。由于 TiDB 能够一次性发送的 job message 默认的最大值为 6 MB（**不建议**修改此值），因此，单次发送的所有表的 schema 总和不可以超过 6 MB，否则，会报 `entry too large, the max entry size is 6291456, the size of data is 7690800` 错误。如需了解有关单个写入请求的数据量的具体限制，请参阅 TiDB 配置项 [txn-entry-size-limit](/tidb-configuration-file.md#txn-entry-size-limit-从-v50-版本开始引入) 和 TiKV 配置项 [raft-entry-max-size](/tikv-configuration-file.md#raft-entry-max-size)。

Reviewer: ```suggestion
    - 解决方法：可以尝试降低并发大小，设置 `--ddl-batch-size=128` 或者更小的值。原因是当 TiDB 执行这个 DDL job 时，TiDB 会把 job 队列写到 TiKV 上。由于 TiDB 能够一次性发送的 job message 默认的最大值为 6 MB（**不建议**修改此值），因此，单次发送的所有表的 schema 总和不可以超过 6 MB，否则，会报 `entry too large, the max entry size is 6291456, the size of data is 7690800` 错误。如需了解有关单个写入请求的数据量的具体限制，请参阅 TiDB 配置项 [txn-entry-size-limit](/tidb-configuration-file.md#txn-entry-size-limit-从-v50-版本开始引入) 和 TiKV 配置项 [raft-entry-max-size](/tikv-configuration-file.md#raft-entry-max-size)。
```

---
title: BR 批量建表
summary: 了解如何使用 BR 批量建表功能。在恢复备份数据时，BR 可以通过批量建表功能加快数据的恢复速度。
---

# BR 批量建表

使用 Backup & Restore (BR) 执行数据恢复任务时，BR 会先在下游 TiDB 集群创建库和表，再进行数据恢复。TiDB v6.0 之前，在数据恢复阶段创建表时，BR 采用了[串行执行](#实现原理)的方案。然而，当需要恢复的数据中带有大量的表（约 50000 张）时，该方案会在创建表上消耗较多时间。

为了通过加快创建表的速度和减少数据恢复的时间，从 TiDB v6.0 起，BR 引入了批量建表功能，此功能默认开启。

> > **注意：**
> >
> > 为使用 BR 建表功能，TiDB 和 BR 都必须为 v6.0 或以上版本。如果 TiDB 或 BR 的任意一方的版本低于 v6.0，BR 会采用原来的串行建表方案。

Reviewer 2: ```suggestion
> **注意：**
>
> - 为使用 BR 建表功能，TiDB 和 BR 都必须为 v6.0 或以上版本。如果 TiDB 或 BR 的任意一方的版本低于 v6.0，BR 会采用原来的串行建表方案。
> - 在使用 v6.0 或以上版本，或从 v6.0 以下版本升级至该版本时，BR 会默认开启批量建表功能，你不需要额外进行相关配置。
```

## 使用场景

当需要恢复的数据中带有大量的表（约 50000 张）时，你可以使用 BR 批量建表功能显著提升数据恢复的速度。

具体提速效果可参考[测试结果](#功能测试)。

## 使用方法

> BR 默认开启了批量建表功能，在 v6.0 或以上版本中默认设置了 `--ddl-batch-size=128`，以加快恢复时的建表速度。因此，你不需要额外配置该参数。

Reviewer 2: ```suggestion
BR 默认开启了批量建表功能，在 v6.0 或以上版本中默认设置了 `--ddl-batch-size=128`（即 BR 会以 128 张表为一批，并发创建多批表），以加快恢复时的建表速度。因此，你不需要额外配置该参数。
```

如果需要关闭此功能，你可以通过以下命令将 `--ddl-batch-size` 的值设置为 `0`：

{{< copyable "shell-regular" >}}

```shell
br restore full -s local:///br_data/ --pd 172.16.5.198:2379 --log-file restore.log --ddl-batch-size=0
```

关闭批量建表功能后，BR 会采用原来的[串行建表方案](#实现原理)。

## 实现原理

- v6.0 前的串行建表方案：在 v6.0 之前的版本，BR 采用了串行建表方案。在使用 BR 执行数据恢复任务时，BR 会先在目标 TiDB 创建库和表后，再开始进行数据恢复。建表时，BR 会在调用 TiDB 接口后，使用 SQL 语句 `Create Table` 创建表。建表任务由 TiDB DDL owner 依次串行执行。在每张表被创建时，各会引起一次 schema 版本的变更，而每次的 schema 版本的变更都需要同步到其他 BR 和其他 TiDB DDL worker。因此，当需要创建的表的数量比较多时，串行建表方案会导致建表时间过长。
- v6.0 起的批量建表方案：批量建表功能采用了并发批量建表方案。从 v6.0 起，在默认情况下，BR 会以 128 张表为一批，并发创建多批表。采用该方案后，BR 每建一批表时，TiDB schema 版本只会变更一次。此方法极大地提高了建表速度。

## 功能测试

以下是在 TiDB v6.0 集群中测试批量建表功能的内容。具体的测试环境如下：

- 集群配置：
    - 15 个 TiKV 实例，每个 TiKV 实例共有 16 个 CPU 核心、80 GB 内存、16 个处理 RPC 请求的线程（即 [`import.num-threads`](/tikv-configuration-file.md#num-threads) = 16）

>     - 3 个 TiDB 实例，每个 TiKV 实例共有 16 个 CPU 核心、32 GB 内存。

Reviewer 2: ```suggestion
    - 3 个 TiDB 实例，每个 TiDB 实例共有 16 个 CPU 核心、32 GB 内存。
```

---
title: BR 批量建表
summary: 了解如何使用 BR 批量建表功能。在恢复备份数据时，BR 可以通过批量建表功能加快数据的恢复速度。
---

# BR 批量建表

> 使用 Backup & Restore (BR) 执行数据恢复任务时，BR 会先在下游 TiDB 集群创建库和表，再进行数据恢复。TiDB v6.0 之前，在数据恢复阶段创建表时，BR 采用了[串行执行](#实现原理)的方案。然而，当需要恢复的数据中带有大量的表（约 50000 张）时，该方案会在创建表上消耗较多时间。

Reviewer 2: 使用 Backup & Restore (BR) 执行数据恢复任务时，BR 会先在目标 TiDB 创建库和表，再进行数据恢复。

> 为了通过加快创建表的速度和减少数据恢复的时间，从 TiDB v6.0 起，BR 引入了批量建表功能，此功能默认开启。

Reviewer 2: 为了加快创建表的速度，以减少数据恢复的时间，从 TiDB v6.0 起，BR 引入了批量建表功能，此功能默认开启。

> **注意：**
>
> - 为使用 BR 建表功能，TiDB 和 BR 都必须为 v6.0 或以上版本。如果 TiDB 或 BR 的任意一方的版本低于 v6.0，BR 会采用原来的串行建表方案。

> > - 在使用 TiUP 等集群管理工具时，v6.0 或以上版本，或从 v6.0 以下版本升级至该版本时，BR 会默认开启批量建表功能，你不需要额外进行相关配置。

Reviewer 2: ```suggestion
> - 如果你在使用 TiUP 等集群管理工具，而且你使用的 TiDB 和 BR 是 v6.0 或以上版本，或是从 v6.0 以下版本升级至该版本，此时，BR 会默认开启批量建表功能，你不需要额外进行相关配置。
```

## 使用场景

当需要恢复的数据中带有大量的表（约 50000 张）时，你可以使用 BR 批量建表功能显著提升数据恢复的速度。

具体提速效果可参考[测试结果](#功能测试)。

## 使用方法

> BR 默认开启了批量建表功能，在 v6.0 或以上版本中默认设置了 `--ddl-batch-size=128`（即 BR 以 128 张表为一批，并发创建多批表），以加快恢复时的建表速度。因此，你不需要额外配置该参数。

Reviewer 3: @fengou1 @IANTHEREAL  `--ddl-batch-size` 的参数说明是否需要也补充到 https://docs.pingcap.com/zh/tidb/dev/use-br-command-line-tool 这篇文档呢？

如果需要关闭此功能，你可以通过以下命令将 `--ddl-batch-size` 的值设置为 `0`：

{{< copyable "shell-regular" >}}

```shell
br restore full -s local:///br_data/ --pd 172.16.5.198:2379 --log-file restore.log --ddl-batch-size=0
```

关闭批量建表功能后，BR 会采用原来的[串行建表方案](#实现原理)。

## 实现原理

> - v6.0 前的串行建表方案：在 v6.0 之前的版本，BR 采用了串行建表方案。在使用 BR 执行数据恢复任务时，BR 会先在目标 TiDB 创建库和表后，再开始进行数据恢复。建表时，BR 会在调用 TiDB 接口后，使用 SQL 语句 `Create Table` 创建表。建表任务由 TiDB DDL owner 依次串行执行。在每张表被创建时，各会引起一次 schema 版本的变更，而每次的 schema 版本的变更都需要同步到其他 BR 和其他 TiDB DDL worker。因此，当需要创建的表的数量比较多时，串行建表方案会导致建表时间过长。

Reviewer 2: - v6.0 前的串行建表方案：在 v6.0 之前的版本，BR 采用了串行建表方案。在使用 BR 执行数据恢复任务时，BR 会先在目标 TiDB 【集群】创建库和表后，再开始进行数据恢复。

> - v6.0 起的批量建表方案：批量建表功能采用了并发批量建表方案。从 v6.0 起，在默认情况下，BR 会以 128 张表为一批，并发创建多批表。采用该方案后，BR 每建一批表时，TiDB schema 版本只会变更一次。此方法极大地提高了建表速度。

Reviewer 2: - v6.0 起的批量建表方案：从 v6.0 起，在默认情况下，BR 会以 128 张表为一批，并发创建多批表。采用该方案后，BR 每建一批表时，TiDB schema 版本只会变更一次。此方法极大地提高了建表速度。

---
title: BR 批量建表
summary: 了解如何使用 BR 批量建表功能。在恢复备份数据时，BR 可以通过批量建表功能加快数据的恢复速度。
---

# BR 批量建表

> 使用 BR 工具（以下简称为 BR）执行数据恢复任务时，BR 会先在下游 TiDB 创建库和表后，再进行数据恢复。TiDB v6.0 之前，在数据恢复阶段创建表时，BR 采用了[串行执行](#实现原理)的方案。然而，当需要恢复的数据中带有大量的表（约 50000 张）时，该方案会在创建表上消耗较多时间。

Reviewer 3: ```suggestion
使用 Backup & Restore (BR) 执行数据恢复任务时，BR 会先在下游 TiDB 集群创建库和表，再进行数据恢复。TiDB v6.0 之前，在数据恢复阶段创建表时，BR 采用了[串行执行](#实现原理)的方案。然而，当需要恢复的数据中带有大量的表（约 50000 张）时，该方案会在创建表上消耗较多时间。
```

> 为了通过加快创建表的速度减少数据恢复的时间，从 TiDB v6.0 起，BR 引入了批量建表功能，此功能会默认开启。

Reviewer 3: ```suggestion
为了通过加快创建表的速度和减少数据恢复的时间，从 TiDB v6.0 起，BR 引入了批量建表功能，此功能默认开启。
```

Reviewer 2: > 
这样修改如何呢？
```suggestion
为了通过加快创建表的速度来减少数据恢复的时间，从 TiDB v6.0 起，BR 引入了批量建表功能，此功能会默认开启。
```

> **注意：**
>
> 为使用 BR 建表功能，TiDB 和 BR 都必须为 v6.0 或以上版本。如果 TiDB 或 BR 的任意一方的版本低于 v6.0，BR 会采用原来的串行建表方案。

## 使用场景

> 当需要恢复的数据中带有大量的表（约 50000 张）时，你可以使用 BR 批量建表功能明显地提升数据恢复速度。

Reviewer 3: ```suggestion
当需要恢复的数据中带有大量的表（约 50000 张）时，你可以使用 BR 批量建表功能显著提升数据恢复的速度。
```

> 具体提速效果，请见[测试结果](#实现原理)。

Reviewer 3: ```suggestion
具体提速效果可参考[测试结果](#实现原理)。
```

## 使用方法

> BR 默认开启了批量建表功能，在 v6.0 或以上版本中默认设置了 `--ddl-batch-size=128`。因此，你不需要额外配置该参数。比如，如果恢复时需要创建 6 万张表，在默认情况下，仅需 5 分钟。

Reviewer 3: + 能否介绍一下 `--ddl-batch-size=128` 参数的用途呢？
+ 如果从 v6.0 以前的集群升到 v6.0，批量建表功能的开启情况是怎样的呢？
+ “比如，如果恢复时需要创建 6 万张表，在默认情况下，仅需 5 分钟。” 这一句是在什么测试环境下达到的数据呢？这里的“默认情况”有种说服力不强感觉😂

Reviewer 2: > * 能否介绍一下 `--ddl-batch-size=128` 参数的用途呢？
> * 如果从 v6.0 以前的集群升到 v6.0，批量建表功能的开启情况是怎样的呢？
> * “比如，如果恢复时需要创建 6 万张表，在默认情况下，仅需 5 分钟。” 这一句是在什么测试环境下达到的数据呢？这里的“默认情况”说服力不强呢😂

最后一条的示例内容，是我总结下面测试部分的测试结果，要不在这里先删掉这句，避免造成内容不清晰的情况吧，因为如果要在这里再列出测试环境的话，和下面的部分就会重复了。

上面两条内容，还请 @fengou1 帮忙补充，感谢！

如果需要关闭此功能，你可以通过以下命令将 `--ddl-batch-size` 的值设置为 `0`：

{{< copyable "shell-regular" >}}

```shell

> bin/br restore full -s local:///br_data/ --pd 172.16.5.198:2379 --log-file restore-concurrency.log --ddl-batch-size=0

Reviewer 3: + 要不要稍加解释下命令里的其他参数呢？
+ 以 `bin/` 开头是预期的吗？

```

关闭批量建表功能后，BR 会采用原来的[串行建表方案](#实现原理)。

## 实现原理

> 在 v6.0 之前的版本，BR 采用了串行建表方案。在使用 BR 执行数据恢复任务时，BR 会先在下游 TiDB 创建库和表后，再开始进行数据恢复。建表时，BR 会在调用 TiDB 接口后，使用 SQL `Create Table` 创建表。建表任务由 TiDB DDL owner 依次串行执行。在每张表被创建时，各会引起一次 schema 版本的变更，而每次的 schema 版本的变更都需要同步到其他 BR 和其他 TiDB DDL worker。因此，当需要创建的表的数量比较多时，串行建表方案会导致建表时间过长。

Reviewer 3: ```suggestion
v6.0 前的串行建表方案：在 v6.0 之前的版本，BR 采用了串行建表方案。在使用 BR 执行数据恢复任务时，BR 会先在下游 TiDB 创建库和表后，再开始进行数据恢复。建表时，BR 会在调用 TiDB 接口后，使用 SQL 语句 `Create Table` 创建表。建表任务由 TiDB DDL owner 依次串行执行。在每张表被创建时，各会引起一次 schema 版本的变更，而每次的 schema 版本的变更都需要同步到其他 BR 和其他 TiDB DDL worker。因此，当需要创建的表的数量比较多时，串行建表方案会导致建表时间过长。
```

> 批量建表功能采用了并发批量建表方案。从 v6.0 起，在默认情况下，BR 会以 128 张表为一批，并发创建多批表。采用该方案后，BR 每建一批表时，TiDB schema 版本只会变更一次。此方法极大地提高了建表速度。

Reviewer 3: ```suggestion
v6.0 起的批量建表方案：批量建表功能采用了并发批量建表方案。从 v6.0 起，在默认情况下，BR 会以 128 张表为一批，并发创建多批表。采用该方案后，BR 每建一批表时，TiDB schema 版本只会变更一次。此方法极大地提高了建表速度。
```

> 以下是在 TiDB v6.0 集群中测试批量建表功能的内容。具体的测试条件如下：

Reviewer 3: 下面的测试属于实现原理吗？这里要不要单独用一个标题来概括下面的测试？

> - 集群配置：
>     - 15 个 TiKV
> 
>         - 有 16 个 CPU 核心、32 GB 内存、16 个处理 RPC 请求的线程（即 [`import.num-threads`](/tikv-configuration-file.md#num-threads) = 16）
> 
>     - 3 个 TiDB
>     - 3 个 PD
> - 待恢复数据的规模：16.16 TB

Reviewer 3: ```suggestion
- 集群配置：

    - 15 个 TiKV

        - 有 16 个 CPU 核心、32 GB 内存、16 个处理 RPC 请求的线程（即 [`import.num-threads`](/tikv-configuration-file.md#num-threads) = 16）

    - 3 个 TiDB
    
    - 3 个 PD
    
- 待恢复数据的规模：16.16 TB
```

> 以下是在 TiDB v6.0 集群中测试批量建表功能的内容。具体的测试条件如下：
> 
> - 集群配置：
>     - 15 个 TiKV
> 
>         - 有 16 个 CPU 核心、32 GB 内存、16 个处理 RPC 请求的线程（即 [`import.num-threads`](/tikv-configuration-file.md#num-threads) = 16）
> 
>     - 3 个 TiDB
>     - 3 个 PD
> - 待恢复数据的规模：16.16 TB
> 

Reviewer 3: ```suggestion
以下是在 TiDB v6.0 集群中测试批量建表功能的内容。具体的测试环境·如下：

- 集群配置：
    - 15 个 TiKV 实例，共有 16 个 CPU 核心、32 GB 内存、16 个处理 RPC 请求的线程（即 [`import.num-threads`](/tikv-configuration-file.md#num-threads) = 16）
    - 3 个 TiDB 实例
    - 3 个 PD 实例
- 待恢复数据的规模：16.16 TB

```

Reviewer 3: 要不要说一下 TiDB、TiKV 等均使用默认配置和变量？

---
title: BR 批量建表
summary: 了解如何使用 BR 工具的批量建表功能。在恢复备份数据时，BR 可以通过批量建表功能加快数据的恢复速度。
---

# BR 批量建表

> 使用 BR 工具（以下简称为 BR）执行数据恢复任务时，BR 会先在下游 TiDB 创建库和表后，再进行数据恢复。在 TiDB v6.0 之前，在数据恢复阶段创建表时，BR 采用了[串行执行](#实现原理)的方案。然而，当需要恢复的数据中带有大量的表（【约 XXX 张】）时，该方案会在创建表时消耗较多时间。

Author: 这里我们用 50000 张吧

为了加快 BR 在执行数据恢复任务时创建表的速度，从而减少数据恢复时间，从 TiDB v6.0 起，BR 引入了批量建表功能，此功能会默认开启。

> **注意：**
>
> 为使用 BR 建表功能，TiDB 和 BR 都必须为 v6.0 或以上版本。如果 TiDB 或 BR 的任意一方的版本低于 v6.0，BR 会采用原来的串行建表方案。

## 使用场景

> 当需要恢复的数据中带有大量的表（【约 XXX 张】）时，使用 BR 批量建表功能可以明显地提升数据恢复速度。具体提升效果，请见[测试结果](#实现原理)。

Author: 大于 50000 张怎么样？

## 使用方法

BR 默认开启了批量建表功能，在 v6.0 或以上版本中默认设置了 `--ddl-batch-size=128`。因此，你不需要额外配置该参数。

- 如果需要关闭此功能，你可以通过以下命令将 `--ddl-batch-size` 的值设置为 `0`：

    {{< copyable "shell-regular" >}}

>     【XXX】

Author: bin/br restore full -s local:///br_data/ --pd 172.16.5.198:2379 --log-file restore-concurrency.log --ddl-batch-size=0

    关闭批量建表功能后，BR 会采用原来的[串行建表方案](#实现原理)。

- 如果你需要恢复的数据中有过多的表（约大于 5 万张），推荐在 BR 中同时配置 `--concurrency=1024` 和 `--ddl-batch-size=256`：

    {{< copyable "shell-regular" >}}

    ```shell
    bin/br restore full -s local:///br_data/ --pd 172.16.5.198:2379 --log-file restore-concurrency.log --concurrency 1024 --ddl-batch-size=256
    ```

>     其中，`--concurrency` 参数用于进行性能调优，默认值为 【`128`（待确认）】。

Author: 确认 默认值是 128

    内部测试显示，在恢复数据时同时配置上述的两个参数后，创建 6 万张表仅需 5 分钟。

## 实现原理

> 在 v6.0 之前的版本，BR 采用了串行建表方案。在使用 BR 执行数据恢复任务时，BR 会先在下游 TiDB 创建库和表后，再开始进行数据恢复。在建表的过程中，BR 调用 TiDB 接口，使用 SQL `Create Table` 创建表，并由 TiDB DDL owner 依次串行执行建表任务。然而，每张表被创建时会引起一次 schema 版本的变更，每次的 schema 变更需要同步到【其他 BR 和其他 TiDB（其他能否用其他更仔细的描述代替呢？）】。该方案会导致在需要创建的表比较多的场景下，建表时间过长。

Author: 内部

> 在 v6.0 之前的版本，BR 采用了串行建表方案。在使用 BR 执行数据恢复任务时，BR 会先在下游 TiDB 创建库和表后，再开始进行数据恢复。在建表的过程中，BR 调用 TiDB 接口，使用 SQL `Create Table` 创建表，并由 TiDB DDL owner 依次串行执行建表任务。然而，每张表被创建时会引起一次 schema 版本的变更，每次的 schema 变更需要同步到【其他 BR 和其他 TiDB（其他能否用其他更仔细的描述代替呢？）】。该方案会导致在需要创建的表比较多的场景下，建表时间过长。

Author: 改为 DDL worker 吧

批量建表功能采用并发批量建表方案。在默认情况下，BR 以 128 张表为一批，并发创建多批表。采用该方案后，BR 每次建一批表时，TiDB schema 版本仅会变更一次。此方法极大地提高了建表速度。

以下为在 TiDB v6.0 集群中测试批量建表功能的测试内容。具体的测试条件如下：

> 【这里可以补充在同等条件下，“未开启功能”时的数据吗？这样用户可以看出功能开启前后的明显的对比】

Author: 成本太高，需要大集群重新做一遍测试。暂时没有办法去拿老版本数据。经验上之前的数据在 50～90 MB/s 之间。
